src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -459,9 +459,7 @@ function assignOptionIds(
     (
       option: OptionDescriptor | ActionListItemDescriptor,
       optionIndex: number,
-    ) => {
-      option.id = `${comboBoxId}-${optionIndex}`;
-    },
+) => ({id: `${comboBoxId}-${optionIndex}`, ...option}),
```

Comment:
Here we're not mutating the option anymore, or using the result of the map so it'll essentially a no-op.

Reply:
@AndrewMusgrave would you be able to run with the appropriate amendments. If this is dependent on me, I doubtful this is going to get fixed. No care if it is my PR that lands. Just wanting the issue fixed.

Reply:
In this fix I would not allow user to set his own id for option as you can see components still relies on his internal id naming in function `private selectedOptionId()`.